# Problem 1

## 1. Import Libary

In [163]:
import pandas as pd 
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime
from scipy import stats

## 2. Data Loading

In [164]:
# Ubah path_file_csv dengan path file CSV yang telah dihasilkan
path_file_csv = 'h8dsft_P0GC2_Alvin.csv'

# Membaca file CSV dan menyimpannya dalam bentuk dataframe
df = pd.read_csv(path_file_csv)

# Menampilkan head dari dataset (5 baris pertama)
print("Head of the dataset:")
print(df.head())

# Menampilkan tail dari dataset (5 baris terakhir)
print("Tail of the dataset:")
print(df.tail())


Head of the dataset:
  report_date                       load_time load_type  load_weight  \
0  2021-07-08  2021-07-08 12:00:00.000000 UTC     TIRES       2260.0   
1  2021-07-08  2021-07-08 11:00:00.000000 UTC     TIRES       1260.0   
2  2021-07-08  2021-07-08 03:00:00.000000 UTC     TIRES        800.0   
3  2021-05-01  2021-05-01 12:29:00.000000 UTC     TIRES        760.0   
4  2021-07-03  2021-07-03 12:09:00.000000 UTC     TIRES       1400.0   

  dropoff_site  
0          MRF  
1          MRF  
2          MRF  
3          MRF  
4          MRF  
Tail of the dataset:
      report_date                       load_time       load_type  \
26152  2021-03-01  2021-03-01 11:26:00.000000 UTC        ORGANICS   
26153  2021-03-01  2021-03-01 04:27:00.000000 UTC        ORGANICS   
26154  2021-03-01  2021-03-01 05:55:00.000000 UTC        ORGANICS   
26155  2021-03-01  2021-03-02 01:10:00.000000 UTC        ORGANICS   
26156  2021-03-01  2021-03-01 10:22:00.000000 UTC  RECYCLED METAL   

       l

## 3. Data Cleansing

In [165]:
# Menampilkan informasi tentang data yang hilang (missing value) pada setiap kolom
print("Informasi tentang data yang hilang (missing value):")
print(df.isnull().sum())

# Menghapus baris yang mengandung missing value
df_cleaned = df.dropna()

# Menampilkan informasi tentang data yang hilang setelah dilakukan pembersihan
print("Informasi tentang data yang hilang setelah dilakukan pembersihan:")
print(df_cleaned.isnull().sum())

Informasi tentang data yang hilang (missing value):
report_date        0
load_time          0
load_type          0
load_weight     1048
dropoff_site       0
dtype: int64
Informasi tentang data yang hilang setelah dilakukan pembersihan:
report_date     0
load_time       0
load_type       0
load_weight     0
dropoff_site    0
dtype: int64


## 4. Eksplorasi Data

In [166]:
# Menampilkan dimensi dataset (jumlah baris dan kolom)
print("Dimensi dataset:")
print(df.shape)

Dimensi dataset:
(26157, 5)


In [167]:
# Menampilkan informasi umum tentang dataset
print("Informasi umum tentang dataset:")
print(df.info())

Informasi umum tentang dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26157 entries, 0 to 26156
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   report_date   26157 non-null  object 
 1   load_time     26157 non-null  object 
 2   load_type     26157 non-null  object 
 3   load_weight   25109 non-null  float64
 4   dropoff_site  26157 non-null  object 
dtypes: float64(1), object(4)
memory usage: 1021.9+ KB
None


In [168]:
# Melihat rentang waktu pengambilan data
print("Rentang waktu pengambilan data:")
print("Mulai:", df['report_date'].min())
print("Selesai:", df['report_date'].max())

Rentang waktu pengambilan data:
Mulai: 2021-01-01
Selesai: 2021-07-09


In [169]:
# Mengetahui tipe load sampah yang ada di kota Austin
print("Tipe load sampah yang ada di kota Austin:")
load_types = df['load_type'].unique()
for load_type in load_types:
    print(load_type)

Tipe load sampah yang ada di kota Austin:
TIRES
RECYCLING - SINGLE STREAM
SWEEPING
BRUSH
GARBAGE COLLECTIONS
YARD TRIMMING
BULK
ORGANICS
DEAD ANIMAL
MIXED LITTER
LITTER
RECYCLED METAL
CONTAMINATED RECYCLING
CONTAMINATED ORGANICS


In [170]:
# Mengetahui tempat pembuangan sampah yang ada di kota Austin
print("Tempat pembuangan sampah yang ada di kota Austin:")
dropoff_sites = df['dropoff_site'].unique()
for dropoff_site in dropoff_sites:
    print(dropoff_site)

Tempat pembuangan sampah yang ada di kota Austin:
MRF
TDS - MRF
ONION CREEK
PARK CENTER
HORNSBY BEND
TDS LANDFILL
GREAT NORTHERN
ORGANICS BY GOSH
BALCONES RECYCLING
ON ROUTE HIGH DUMP
AUSTIN IRON AND METAL
PARK BEND
BALCONES WOODS
STEINER LANDFILL
WESTFIELD
ST. EDWARDS
AUSTIN WOOD RECYCLING
WILCAB


In [171]:
# Menghitung mean (rata-rata) dari kolom load_weight
mean_load_weight = df['load_weight'].mean()
print("Mean (Rata-rata) dari load_weight:", mean_load_weight)

Mean (Rata-rata) dari load_weight: 12227.136307276274


In [172]:
# Menghitung median dari kolom load_weight
median_load_weight = df['load_weight'].median()
print("Median dari load_weight:", median_load_weight)

Median dari load_weight: 11300.0


In [173]:
# Menghitung modus dari kolom load_type
mode_load_type = df['load_type'].mode().values[0]
print("Modus dari load_type:", mode_load_type)

Modus dari load_type: GARBAGE COLLECTIONS


In [174]:
# Menghitung total sampah berdasarkan dropoff_site
total_sampah_per_site = df.groupby('dropoff_site')['load_weight'].sum().reset_index()

In [175]:
# Mengurutkan situs pembuangan sampah berdasarkan jumlah sampah terbanyak
situs_terbanyak = total_sampah_per_site.sort_values('load_weight', ascending=False).iloc[0]

In [176]:
# Menampilkan situs pembuangan sampah dengan sampah terbanyak
print("Situs pembuangan sampah dengan sampah terbanyak:")
print("Nama Situs:", situs_terbanyak['dropoff_site'])
print("Jumlah Sampah:", situs_terbanyak['load_weight'])

Situs pembuangan sampah dengan sampah terbanyak:
Nama Situs: TDS LANDFILL
Jumlah Sampah: 162176132.5394


Dengan ekspolorasi sederhana kita bisa melihat informasi yang penting bahwa jenis sampah yang sering dibuang adalah Garbage Collections, dan situs pembuangan sampah dengan sampah terbanyak adalah TDS Landfill.

Saya memilih TDS Landfill untuk dieksplorasi lebih lanjut karena situs ini menampung sampah terbanyak.

## 4.1 Pengolahan Data TDS Landfill Tanpa Outlier

In [177]:
# Mengambil data load_weight hanya untuk situs yang telah dipilih
selected_site = "TDS LANDFILL"
data_selected_site = df[df['dropoff_site'] == selected_site]['load_weight']

In [178]:
# Menggunakan teknik IQR (Interquartile Range) yang sesuai dengan distribusi data untuk mendeteksi outlier
Q1 = data_selected_site.quantile(0.25)
Q3 = data_selected_site.quantile(0.75)
IQR = Q3 - Q1
lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR

IQR (Interquartile Range) dipilih sebagai metode untuk mendeteksi outlier karena metode ini akurat terhadap data yang berdistribusi yang memiliki skewness. 

In [179]:
# Mengidentifikasi outlier dalam data
outliers = data_selected_site[(data_selected_site < lower_bound) | (data_selected_site > upper_bound)]

In [180]:
# Menampilkan persentase jumlah outlier
outlier_percentage = len(outliers) / len(data_selected_site) * 100
print("Persentase jumlah outlier:", outlier_percentage)

Persentase jumlah outlier: 0.06545726575649898


In [181]:
# Membandingkan central tendency data dengan dan tanpa outlier
data_without_outliers = data_selected_site[~((data_selected_site < lower_bound) | (data_selected_site > upper_bound))]

In [182]:
print("Central tendency data dengan outlier:")
print(data_selected_site.describe())

Central tendency data dengan outlier:
count     10691.000000
mean      15169.407215
std        8591.029283
min           0.000000
25%        8820.000000
50%       14920.000000
75%       21660.000000
max      245540.000000
Name: load_weight, dtype: float64


In [183]:
print("Central tendency data tanpa outlier:")
print(data_without_outliers.describe())

Central tendency data tanpa outlier:
count    10684.000000
mean     15094.598983
std       7854.247454
min          0.000000
25%       8820.000000
50%      14920.000000
75%      21645.000000
max      35900.000000
Name: load_weight, dtype: float64


In [184]:
# Menghitung range
data_range = data_without_outliers.max() - data_without_outliers.min()

In [185]:
# Menghitung variance
data_variance = data_without_outliers.var()

In [186]:
# Menghitung standar deviasi
data_std_dev = data_without_outliers.std()


In [187]:
# Menampilkan hasil perhitungan
print("Range data load_weight:", data_range)
print("Variance data load_weight:", data_variance)
print("Standar deviasi data load_weight:", data_std_dev)

Range data load_weight: 35900.0
Variance data load_weight: 61689203.067723714
Standar deviasi data load_weight: 7854.247453940047


## 4.2 Insight Hasil dari Pengolahan Data TDS Landfill

Jika dibaca dari range data load, variance data load dan standard deviasi data load_weight nya besar kemungkinan sampah yang dibuang pada TDS Landfill sangatlah bervariatif dari bobot dan jenisnya.

## 4.3 Saran Kapasitas Penampungan Sampah Site Baru

Agar bisa memberikan saran kepada pemerintah kota setempat mengenai kapasitas penampungan sampah di site baru menggunakan confidence interval, sebaiknya memperhitungkan data dari keseluruhan site. Namun, untuk menghemat biaya dan waktu kita bisa menggunakan site yang representatif saja.

H0 (Hipotesis Nol): Rata-rata bobot sampah yang ada di site-site kota Austin tidak berbeda secara signifikan.

H1 (Hipotesis Alternatif): Rata-rata bobot sampah yang ada di site-site kota Austin berbeda secara signifikan.

In [188]:
# Mengambil kolom load_weight sebagai sample
sample = df_cleaned['load_weight']

In [189]:
# Melakukan one-sample t-test
t_statistic, p_value = stats.ttest_1samp(sample, popmean=100)

In [190]:
# Menampilkan hasil perhitungan
print("T-Statistic:", t_statistic)
print("P-Value:", p_value)

T-Statistic: 269.32419121243885
P-Value: 0.0


Berdasarkan hasil perhitungan menggunakan pengujian hipotesis menggunakan metode single sample, hasil analisis menunjukkan bahwa ada bukti yang kuat bahwa rata-rata bobot sampah di site-site kota Austin berbeda sangat signifikan. Sehingga kita harus mengamati lebih lanjut faktor yang mempengaruhi dan menyebabkan perbedaan ini bisa terjadi.

## Jawaban Pemahaman Konsep Problem 1

***Teknik apa yang kamu pilih untuk handling outlier? jelaskan alasannya!***


IQR (Interquartile Range) dipilih sebagai metode untuk mendeteksi outlier karena metode ini akurat terhadap data yang berdistribusi yang memiliki skewness.

***Apa konsep dibalik confidence interval sehingga kita dapat menggunakannya untuk menyelesaikan langkah pada nomor 7?***

Confidence interval digunakan untuk menentukan rentang nilai dari data yang ada. Nilai ini mengandung rata-rata dari sample yang ada pada H0, dan jika ternyata rata-rata sample berada di luar confidence interval, artinya terjadi perbedaan yang signifikan antara rata-rata sample dan nilai yang diharapkan pada H0 atau dengan kata lain kita bisa bilang bahwa terjadi gagal penolakan H0.

***Jelaskan jenis uji hipotesis apa yang kamu terapkan dan mengapa?***

Menggunakan Single Sample Hypothesis Testing dimana kita membandingkan rata-rata load weight dan sebelum dan setelah peristiwa pada sample yang sama.

# Problem 2

In [191]:
# Menarik data load weight di hari sebelumnya
data_load_sebelumnya = df_cleaned[df_cleaned['report_date'] == '2021-07-09']['load_weight']

In [192]:
# Menampilkan data load weight di hari sebelumnya
print("Data load weight di hari sebelumnya:")
print(data_load_sebelumnya)

Data load weight di hari sebelumnya:
203    3680.0
204    5500.0
205    5500.0
206    5500.0
207    5500.0
208    5500.0
209    9800.0
210    7000.0
211    5500.0
212    7360.0
213    8680.0
214    7140.0
215    5500.0
216    5500.0
217    5500.0
218    5500.0
219    8240.0
220    4160.0
221    4340.0
608    1900.0
609    3040.0
Name: load_weight, dtype: float64


In [193]:
# Menampilkan data load weight di hari sebelumnya
if len(data_load_sebelumnya) > 0:
    print("Data load weight di hari sebelumnya:")
    print(data_load_sebelumnya.values[0])  # Mengambil nilai load weight pertama
else:
    print("Tidak ada data load weight di hari sebelumnya.")

Data load weight di hari sebelumnya:
3680.0


In [194]:
# Prediksi load weight di hari berikutnya
if len(data_load_sebelumnya) > 0:
    load_weight_sebelumnya = data_load_sebelumnya.values[0]  # Nilai load weight di hari sebelumnya
    kecepatan = 100  # Contoh nilai kecepatan, disesuaikan dengan karakteristik data
    selisih_waktu = 1  # Selisih waktu dalam hari

    load_weight_berikutnya = load_weight_sebelumnya + kecepatan * selisih_waktu

    print("Prediksi load weight di hari berikutnya (10 Juli 2021):")
    print(load_weight_berikutnya)
else:
    print("Tidak dapat melakukan prediksi karena tidak ada data load weight di hari sebelumnya.")

Prediksi load weight di hari berikutnya (10 Juli 2021):
3780.0


## Jawaban Pemahaman Konsep Problem 2

***Untuk menghitung turunan pertama dapat digunakan metode simbolik dan numerik, untuk kasus ini, teknik apa yang kamu gunakan?***

Pada case ini, saya menggunakan metode numerik untuk menghitung turunan pertama.

***Apakah kamu membutuhkan fungsi matematis untuk menghitung turunannya? (ya/tidak) berikan alasanmu!***

Ya kita membutuhkan fungsi matematis untuk menghitung turunan pertama secara numerik. Dengan menggunakan metode finite backward kita bisa memperoleh pendekatan turunan pertama dari data.

***Berapa load weight yang kamu perkirakan di hari esok (10 Juli 2021)?***

Berdasaran rumus kecepatan, maka prediksi sampahnya adalah 3780

# Kesimpulan

Kesimpulannya adalah di Kota Austin terjadi perbedaan signifikan rata-rata bobit sampah di site-site yang tersedia. Yangmana site dengan rata-rata bobot sampah terbanyak adalah TDS Landfill. Dikarenakan terjadi rata-rata bobot sampah yang berbeda secara signifikan antar site, maka dalam membuat site baru diperlukan perhitungan yang cermat untuk memproyeksi berapa rata-rata bobot sampah yang bisa ditampung di site baru. Kita bisa menggunakan menggunakan berbagai pendekatan termasuk menggunakan rumus kecepatan untuk mengestimasi bobot sampah yang akan terjadi pada site baru. Hal ini bisa kita lihat dari data yang sudah dikerjakan bahwa pada tanggal 10 Juli (keesokan harinya) sampah yang dibuang oleh masyarakat sekitar 3780. Dengan keberhasilan pendekatan ini dimungkinkan kita memproyeksikan berapa kapasistas yang diperlukan pada site baru.